kbfscrypto/auth_token.go

```diff
@@ -50,14 +56,15 @@ func NewAuthToken(config Config, tokenType string, expireIn int,
 // Sign is called to create a new signed authentication token.
 func (a *AuthToken) signWithUserAndKeyInfo(ctx context.Context,
 	challengeInfo keybase1.ChallengeInfo, uid keybase1.UID,
-	username libkb.NormalizedUsername, key VerifyingKey) (string, error) {
+	username libkb.NormalizedUsername,
+	key VerifyingKey) (string, error) {
 	// create the token
-	token := auth.NewToken(uid, username, key.kid, a.tokenType,
+	token := auth.NewToken(uid, username, key.KID(), a.tokenType,
 		challengeInfo.Challenge, challengeInfo.Now, a.expireIn,
 		a.clientName, a.clientVersion)
 
 	// sign the token
-	signature, err := a.config.Crypto().SignToString(ctx, token.Bytes())
+	signature, err := a.signer.SignToString(ctx, token.Bytes())
```

Comment:
Ugh, yeah. The initialization order in init.go was off; mdserver was being created before crypto was, but it wasn't showing up before since we were passing in a config.